src/thunderbird_accounts/authentication/tests.py

```diff
@@ -334,3 +337,140 @@ def test_fxa_process_delete_user(self):
+            data={'email': email, 'secret': self.client_env.auth_token},
+        )
+        result = response.json().get('result')
+
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+    def test_is_a_user(self):
+        response = self.api_client.post(
+            'http://testserver/api/v1/auth/is-in-allow-list/',
+            data={'email': self.user.email, 'secret': self.client_env.auth_token},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        result = response.json().get('result')
+
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+    def test_is_not_in_list(self):
+        settings.FXA_ALLOW_LIST = '@example.org'
+        email = 'greg@example.com'
+
+        response = self.api_client.post(
+            'http://testserver/api/v1/auth/is-in-allow-list/',
+            data={'email': email, 'secret': self.client_env.auth_token},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        result = response.json().get('result')
+
+        self.assertIsNotNone(result)
+        self.assertFalse(result)
+
+    def test_entry_is_cached(self):
+        """Test that the entry is cached, and is deleted on user delete"""
+        response = self.api_client.post(
+            'http://testserver/api/v1/auth/is-in-allow-list/',
+            data={'email': self.user.email, 'secret': self.client_env.auth_token},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        result = response.json().get('result')
+
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+        self.assertTrue(get_cache_allow_list_entry(self.user.email))
+
+        # okay lets delete the user, this should remove the cache...
+        self.user.delete()
+
+        self.assertIsNone(self.user.uuid)
+
+        self.assertFalse(get_cache_allow_list_entry(self.user.email))
+
+    def test_entry_with_updated_email_is_uncached(self):
+        """If a user's email changes via a fxa webhook, we should make sure the cache is cleared for that entry."""
+        settings.FXA_ALLOW_LIST = '@example.com'  # Not .org like the original email!
+        original_email = self.user.email
+        new_email = 'new-email@example.com'
+
+        # Wouldn't normally be in the allow list
+        self.assertFalse(original_email.endswith(settings.FXA_ALLOW_LIST))
+
+        response = self.api_client.post(
+            'http://testserver/api/v1/auth/is-in-allow-list/',
+            data={'email': self.user.email, 'secret': self.client_env.auth_token},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        result = response.json().get('result')
+
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+        self.assertTrue(get_cache_allow_list_entry(self.user.email))
+
+        # okay lets delete the user, this should remove the cache...
```

Comment:
nit, comment should be something like `okay lets change the user's email`

Reply:
Copy-pasta error, thanks for the catch!